GDJS/GDJS/IDE/ExporterHelper.cpp

```diff
@@ -277,16 +281,39 @@ bool ExporterHelper::ExportCordovaFiles(const gd::Project &project,
           .FindAndReplace("<!-- GDJS_ICONS_ANDROID -->", makeIconsAndroid())
           .FindAndReplace("<!-- GDJS_ICONS_IOS -->", makeIconsIos());
 
-  if (!project.GetAdMobAppId().empty()) {
-    str = str.FindAndReplace(
-        "<!-- GDJS_ADMOB_PLUGIN_AND_APPLICATION_ID -->",
-        "<plugin name=\"cordova-plugin-admob-free\" spec=\"~0.21.0\">\n"
-        "\t\t<variable name=\"ADMOB_APP_ID\" value=\"" +
-            project.GetAdMobAppId() +
-            "\" />\n"
-            "\t</plugin>");
+  gd::String plugins = "";
+
+  for(gd::String const extensionName : project.GetUsedExtensions()) {
+    std::shared_ptr<gd::PlatformExtension> extension = project.GetCurrentPlatform().GetExtension(extensionName);
+
+    for (gd::DependencyMetadata dependency: extension->GetAllDependencies()) {
+      //if (dependency.GetDependencyType() == gd::DependencyTypes::cordova) {
+
+        plugins += "<plugin name=\"" + dependency.GetExportName();
+        if(dependency.GetVersion() != "-1") {
+          plugins += "\" spec=\"" + dependency.GetVersion();
+        }
+        plugins += "\">\n"; 
+
+        // In cordova all settings are considered a plugin variable
+        for (std::pair<gd::String, gd::PropertyDescriptor>& variable : dependency.GetAllExtraSettings()) {
+          if(variable.second.GetType() == "ProjectProperty") {
+            plugins += "\t\t<variable name=\"" + variable.first + "\" value=\"" + project["Get" + variable.second.GetValue()]() + "\" />\n";
```

Comment:
Unless I missed something you cannot generate method name and call them like this dynamically in C++.

I think this should work a bit in the other way: 
* ExporterHelper should iterate on all extensions (ideally only those used by the project, but we'll fix that later) and for each extension:
* It should call a method on the extension, **passing** to it the project (`extension->GenerateDependencies(project)`, `extension->ExposeDependencies(project)`, something like that).
* This method is responsible for looking in the project if there is the required configuration somewhere (for example, the AdMob plugin stores it in `project.GetAdMobAppId`.. but we could create a generic storage for parameters of extensions :) AdMob is a spoiled child to which I give a special place in gd::Project - I should not have ;)).
* The ExporterHelper will get the result and write it (in the config.xml for Cordova, but could be something else in the future for another platform).

In short, you're telling to the extension: "hey extension, please give me the list of your dependencies. To help you, I pass you the project - you'll find your settings somewhere inside, but I don't know where - it's your problem as an extension".

(Though the "your problem as an extension" is a bit hard because extensions can't just store stuff in gd::Project! Hence why we'll need to add a `ExtensionSpecificProperties` class to `gd::Project` one day)
(and finally, something to allow the extension to declare the `properties` that should be shown in the IDE and stored in `ExtensionSpecificProperties`).


Hope this kind of make sense :)

Reply:
> ExtensionSpecificProperties 
Huh I was thinking of doing that in a future PR so ok.

> Hope this kind of make sense :)
Yes, this makes total sense. Just one thing I am unsure about: I don't think it is possible to overwrite WebAssembly object functions. This could be an issue.

Reply:
> Huh I was thinking of doing that in a future PR so ok.

Yes, I think that this will need to be splitted in multiple PRs.

> Just one thing I am unsure about: I don't think it is possible to overwrite WebAssembly object functions. This could be an issue.

It's possible, given a base class, to create an object that overrides a function, but it's a bit complex. We'll see if we really need this. Maybe we can have default functions for Cordova/npm, so that 99% of extensions could just call a function to register their Cordova/npm dependencies. After all, let's try to keep stuff simple rather than over-engineering a "generic" way of adding dependencies that risk to be never used for anything else than Cordova/npm :)